themes/default/content/blog/pulumi-developer-workflow/index.md

```diff
@@ -0,0 +1,409 @@
+cp -f .devcontainer/devcontainer.json .devcontainer.json
+
+# Rebuild the codespaces using the newly added Pulumi Devcontainer
+gh codespace rebuild --codespace ${CODESPACE_NAME}
+```
+
+#### Pulumi Cloud
+
+Let's login to Pulumi Cloud and initialize a new ESC environment to store our environment variables, secrets, and configuration which should be preserved and distributed securely from your Pulumi Cloud account.
+
+```bash
+# Login to Pulumi Cloud
+pulumi login
+
+# Create a new ESC Environment (optionally use another already existing environment)
+pulumi env init workshop
+```
+
+#### KinD: Kubernetes-in-docker
+
+We are going to use Kubernetes to demonstrate our Pulumi IaC. Let's go ahead and create a new cluster now.
+
+```bash
+# Create the directory for your new kind kubeconfig
+mkdir .kube
+
+# Create a Kind Cluster
+kind create cluster
+
+# Let's make sure our kubeconfig works and pods are starting up
+kubectl get pods -A
+
+# Load Kubeconfig into Pulumi ESC as an encrypted secret for safe keeping.
+# This is an example use case for storing kubeconfig which would not usually be practical
+# for disposable kind kubernetes clusters like this however the approach is practical in many
+# other scenarios which you may come across.
+pulumi env set workshop secrets.kubeconfig.kind --secret "$(jq . -R -s < $KUBECONFIG)"
+pulumi env set workshop files.KUBECONFIG --plaintext \${secrets.kubeconfig.kind}
+```
+
+In our environment, we maintain our secrets including API personal access tokens, kubeconfigs, and such all with Pulumi's Environments, Secrets, and Configuration cloud service.
+
+```bash
+# Load the newly updated Pulumi ESC Environment in the local shell
+eval $(pulumi env open workshop --format shell)
+```
+
+#### Now let's create the deployment code!
+
+If fortune favors the bold, let's be bold on this next step and let AI write our sample code on the fly!
+
+As this is a "hello world" style demonstration intended to showcase the developer workflow more than any specific cloud technologies, let's deploy something fun like a Minecraft server! Be sure to watch the video demonstration of this exercise on Pulumi's [PulumiTV](https://youtube.com/@PulumiTV) YouTube channel for an extra easter egg bonus step!
+
+```bash
+# Write a new Pulumi Python IaC program to deploy Minecraft on Kubernetes
+pulumi new \
+  --ai "Write a program using pulumi kubernetes helm v3 Release to deploy the itzg/minecraft-server helm chart on Kubernetes, and include a pulumi output to show the helm release status." \
+  --description "A pulumi infrastructure as code (iac) program for deploying and serving minecraft on kubernetes" \
+  --name "minecraft-on-kubernetes" \
+  --language python \
+  --stack "workshop" \
+  --force \
+  --dir .
+```
+
+Now let's see if our new Infrastructure as Code Pulumi Python Codebase will deploy!
+
+```bash
+# Deploy your new Pulumi IaC on Kubernetes
+pulumi stack select workshop
+pulumi up
+```
+
+#### Conclusion
```

Comment:
Seems to me like we should pull the Conclusion section out of the details and show it all the time (not only when Details is expanded).